### ISSUE TITLE
screen keeps turning on, keeping it from working properly

### ISSUE THREAD

**Issue by nathgit** [scheibler/TactileClock#1]
For some reason, when I try to activate this, Assistant keeps getting triggered instead, which turns the phone on & keeps it from working. I have the click internal set to 250 & 300. Is that too short or too long or is there something else I can do to get it to work. I don't know if assistant is what's turning the screen, or if it's something else. I have it clipped on to my hip, so the pricing sensor shouldn't be to blame, should it? I turned off the setting in assistant to trigger on the lock screen, so it shouldn't be starting anyway.

**Comment by scheibler** [comment]
I can only guess but I think that you double click too fast. At least 300 milliseconds as the upper boundary
is not enough. Default is 1350 ms.

Unfortunately Android doesn't let me detect the power button clicks directly. To solve this I listen
for screen on / off instead. Now if you double click the power button too fast, the latter one which
should turn off the screen, is missed by the system and therefore by my app too. This results in
your screen staying on.

Please try a click interval between 250 and at least 1000 ms. Better make it 1500 ms at first. And
then click more slowly, on..off. I admit that it takes some practice and it's a bit of a hack but I
don't think, that there is a better solution for it.

I don't know, why the assistant should start up, I don't use it. But I only found one setting
regarding power button and assistant in my phones settings ant that's about power button long clicks
(decide, if the on/off menu or the assistant should be shown on long click).  And if my speculations
above are not completely off then this setting couldn't be it...

And just to be on the safe site make sure, that you've enabled the power button click function in the app at all.

**Comment by nathgit** [comment]
I finally got it working most of the time, by setting it to 1000 & 2000. It's still a little touchy occasionally, but it works most of the time. Thanks for the help. My watch is always a couple minutes slow, so I rely on my phone most of the time, so it's nice to be able to get the time without having to take it off my hip to look. I set assistant to none, so it doesn't come up now. I don't use it either.
